### PR TITLE
Hardlight Spear traitor item

### PR DESCRIPTION
**Pull request by Oricana-16**
# About The Pull Request

Adds the Hardlight Spear implant, a 10 tc traitor item that lets you summon a spear every 30 seconds. The spear deals quite a bit of damage, and disappears once it gets thrown (whether you hit or not, there are some cases where the spear might stay though, usually when you get the killing blow on something). If you use it as a melee weapon, it does slightly more damage than a normal spear, but much less than throwing it does.

## Why It's Good For The Game

Adds a more conspicuous (or inconspicuous depending on how you play it.), big damage throwing item for traitors, that you dont need to/cant spam with at all, and also energy spears are just sick af.

## Testing Photographs and Procedure
<details>
<summary>Screenshots&Videos</summary>

HoldingPhoto

ImplantPhoto

UplinkPhoto

</details>

## Changelog

:cl:
add: Added Hardlight Spear implant (10tc) for traitors.
imageadd: added a hardlight spear icon for both an implant button icon and an item icon
/:cl:

### CODE REVIEW

File: code/game/objects/items/devices/sound_synth.dm [hunk 1]
```diff
@@ -1,6 +1,6 @@
 /*
         Ported from /vg/station:
-        https://github.com/vgstation-coders/vgstation13/blob/Bleeding-Edge/code/game/objects/items/devices/sound_synth.dm
+        https://github.com/vgstation-coders/vgstation13/blob/Bleeding-Edge/code/game/objects/items/devices/soundsynth.dm
```
Comment (nednaZ) [inline code/game/objects/items/devices/sound_synth.dm:3]: Why was this edited?

File: monkestation/code/game/objects/items/implants/implant_hardspear.dm [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+	implant_color = "b"
+	uses = -1
+	COOLDOWN_DECLARE(hardlight_implant_cooldown)
+
+/obj/item/implant/hardspear/activate()
+	if(!COOLDOWN_FINISHED(src, hardlight_implant_cooldown)) //Thanks implant_abductor.dm for the help <3
+		to_chat(imp_in, "<span class='warning'>You must wait [COOLDOWN_TIMELEFT(src, hardlight_implant_cooldown)*0.1] seconds to use [src] again!</span>")
+		return
+
+	var/obj/item/spear/awesomespear
```
Comment (nednaZ) [inline monkestation/code/game/objects/items/implants/implant_hardspear.dm:14]: You can use the typepath of the hard light spear directly.
new var/obj/item/spear/hardlightspear/summoned_spear(loc) for example.

Also, change the variable name to something more relevant for readability

File: monkestation/code/game/objects/items/implants/implant_hardspear.dm [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+/obj/item/implanter/hardspear
+	name = "implanter (hardlight spear)"
+	imp_type = /obj/item/implant/hardspear
+
+/obj/item/implantcase/hardspear
+	name = "implant case - 'Hardlight Spear'"
+	desc = "A glass case containing a freedom implant."
+	imp_type = /obj/item/implant/hardspear
+
+/obj/item/spear/hardlightspear
```
Comment (nednaZ) [inline monkestation/code/game/objects/items/implants/implant_hardspear.dm:33]: I prefer snake_case

File: monkestation/code/game/objects/items/implants/implant_hardspear.dm [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+		qdel(awesomespear) //not so awesome anymore, are you buddy?
+		return
+
+/obj/item/implanter/hardspear
+	name = "implanter (hardlight spear)"
+	imp_type = /obj/item/implant/hardspear
+
+/obj/item/implantcase/hardspear
+	name = "implant case - 'Hardlight Spear'"
+	desc = "A glass case containing a freedom implant."
```
Comment (nednaZ) [inline monkestation/code/game/objects/items/implants/implant_hardspear.dm:30]: Inaccurate description

File: monkestation/code/game/objects/items/implants/implant_hardspear.dm [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+	armour_penetration = 18
+	hitsound = 'sound/weapons/blade1.ogg'
+	attack_verb = list()
+	sharpness = IS_SHARP
+
+/obj/item/spear/hardlightspear/throw_impact(atom/hit_atom, datum/thrownthing/throwingdatum)
+	. = ..()
+	if(hit_atom && !QDELETED(hit_atom))
+		SEND_SIGNAL(src, COMSIG_MOVABLE_IMPACT, hit_atom, throwingdatum)
+		if(is_hot() && isliving(hit_atom))
```
Comment (nednaZ) [inline monkestation/code/game/objects/items/implants/implant_hardspear.dm:58]: Why is it checking for heat?

File: monkestation/code/game/objects/items/implants/implant_hardspear.dm [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+	COOLDOWN_DECLARE(hardlight_implant_cooldown)
+
+/obj/item/implant/hardspear/activate()
+	if(!COOLDOWN_FINISHED(src, hardlight_implant_cooldown)) //Thanks implant_abductor.dm for the help <3
+		to_chat(imp_in, "<span class='warning'>You must wait [COOLDOWN_TIMELEFT(src, hardlight_implant_cooldown)*0.1] seconds to use [src] again!</span>")
+		return
+
+	var/obj/item/spear/awesomespear
+	awesomespear = new /obj/item/spear/hardlightspear(imp_in.loc)
+	if(imp_in.put_in_hands(awesomespear))
```
Comment (nednaZ) [inline monkestation/code/game/objects/items/implants/implant_hardspear.dm:16]: You can have this proc delete the spear on fail

**Review by nednaZ — CHANGES_REQUESTED** [review]
Looks like it will be a fun item, just needs some work.

**Review by nednaZ — CHANGES_REQUESTED** [review]
I like the overall design, though it strangely feels weak.
I'd like a sound effect that plays upon summoning the spear and maybe a reduction in cooldown to 20 seconds.

**Comment by Oricana-16** [comment]
> I'd like a sound effect that plays upon summoning the spear

Do you have any recommendations for the sound? I think the e-sword turning on (saberon.ogg) seems fine, but I'm not sure if that'd be too confusing or anything if they use the same sound.

**Comment by Oricana-16** [comment]
I lowered the cooldown and made the sound play. If you think the sound should be different lmk but otherwise 👍.

**Review by nednaZ — APPROVED** [review]
I like it.
Maybe in the future it will need tweaking, but I think it's good enough to use for now.

**Pull request merged**